### DISCUSSION TITLE
Poetry install packages but doesn't found module

### DISCUSSION THREAD

**Discussion by danilobatistaqueiroz** [python-poetry/poetry#4305] [Q&A] [answered]
I installed poetry in my Ubuntu, started a new project using **poetry new poetry-demo**, but when I install a package, and start using it, Python can't find the module.

The same problem occurs with pipenv.

Ubuntu 20.04

I'm using my user and not root.

I tried to install camelcase package:

inside poetry.lock

```
[[package]]
name = "camelcase"
version = "0.2"
description = "Converts a string to Camel Case"
category = "main"
optional = false
python-versions = "*"
```

inside pyproject.toml

```
[tool.poetry.dependencies]
python = "^3.8"
camelcase = "^0.2"
```

inside cameltest.py
```
import camelcase
```

running `poetry install` the screen shows everything ok:

![](https://i.stack.imgur.com/wXDaD.png)

running `poetry run python3.8 camelcases.py` shows:
```
Traceback (most recent call last):
  File "camelcases.py", line 1, in <module>
    import camelcase
ModuleNotFoundError: No module named 'camelcase'
```

![](https://i.stack.imgur.com/yWqDg.png)

If I run `ls /home/danilo/.cache/pypoetry/virtualenvs/poetry-demo-NpfZs6bz-py3.8/lib/python3.8/site-packages` 

it lists many files and directories, but doesn't show camelcase module.

if I run `poetry run pip list` it doesn't list camelcase module.

I ran an ubuntu docker container, installed poetry, and its fine, the directory `/home/danilo/.cache/pypoetry/virtualenvs/poetry-demo-NpfZs6bz-py3.8/lib/python3.8/site-packages` and the command `poetry run pip list` list the camelcase module, but in my machine, it doesn't work.

My default Python3 is 3.8 : `python3 --version`  
But when I use `poetry install` the **camelcase** package is installed inside my **python3.6** installation outside virtualenv, `/home/danilo/.local/lib/python3.6/site-packages/camelcase`

**Comment by danilobatistaqueiroz** [comment, marked as answer]
The problem is pip, it installs in python3.6 folder, and when it tries to uninstall, it searches in python3.8 folder

**Reply by danilobatistaqueiroz** [reply]
it was the global configuration of pip

**Reply by maxzw** [reply]
How did you solve this?

**Reply by lejshi3** [reply]
![image](https://github.com/python-poetry/poetry/assets/45546025/872ca05d-80e9-40ad-b1f5-84efa6427e70)

**Reply by 0xUgochukwu** [reply]
in case you're still having this problem in poetry you can turn on the option to create a virtual environment in poetry before installing the packages, also don't forget to activate the virtual environment or add the path to your `PATH` environment variable

**Comment by Suvorrow** [comment]
I had the similar problem. I solved it simply: 
**$ poetry lock** 
My project was just starting, so I didn't use the --no-update flag
and after that the command:
**$ poetry install**
